core/transaction.go

```diff
@@ -409,6 +422,38 @@ func transactionCommitment(transactions []Transaction) (*felt.Felt, error) {
 	})
 }
 
+// blockVersionPost011 determines if the blocks protocolVersion is >=0.11.1.
+func blockVersionPost011(protocolVersion string) (bool, error) {
+
+	versionConstraint, _ := semver.NewConstraint(">=0.11.1")
+	var blockVersion *semver.Version
+	var err error
+
+	if protocolVersion == "" {
+		blockVersion, _ = semver.NewVersion("0.0.0")
+	} else {
+		blockVersion, err = parseBlockVersion(protocolVersion)
+		if err != nil {
+			return false, err
+		}
+	}
+	return versionConstraint.Check(blockVersion), nil
+}
+
+func parseBlockVersion(version string) (*semver.Version, error) {
```

Comment:
please use this function in https://github.com/NethermindEth/juno/blob/c3e110e2823b1ee80d1136a53b657b6c273f7068/blockchain/blockchain.go#L44-L65 as well

Reply:
Importing the blockchain package into core creates a cyclic dependency (which is why I effectively copied the checkBlockVersion logic into parseBlockVersion. Do you have any suggestions on how I might resolve this cleanly?